Trading/Mode/dip_analyser_trading_mode.py

```diff
@@ -0,0 +1,397 @@
+
+    async def create_state(self):
+        if self.first_trigger:
+            # can't rely on previous execution buy orders: need plans for sell orders
+            await self._cancel_buy_orders()
+            self.first_trigger = False
+        if self.final_eval != START_PENDING_EVAL_NOTE:
+            self.volume_weight = self.final_eval["volume_weight"]
+            self.price_weight = self.final_eval["price_weight"]
+            await self._create_bottom_order(self.final_eval["current_candle_time"])
+        else:
+            self.volume_weight = self.price_weight = None
+
+    async def order_filled_callback(self, filled_order):
+        if filled_order.get_side() == TradeOrderSide.BUY:
+            order_identifier = self._get_order_identifier(filled_order)
+            if order_identifier in self.sell_targets_by_order:
+                sell_target = self.sell_targets_by_order[order_identifier]
+                trader = filled_order.trader
+                sell_quantity = filled_order.get_filled_quantity() - filled_order.get_total_fees(self.quote)
+                buy_price = filled_order.get_origin_price()
+                await self._create_order(trader, False, sell_quantity, sell_target, buy_price)
+            else:
+                self.logger.error(f"No sell target from order {filled_order}. Can't create sell orders.")
+
+    async def _create_bottom_order(self, notification_candle_time):
+        self.logger.info(f"** New buy order for ** : {self.symbol}")
+        # call orders creation method
+        if self.symbol_evaluator.has_trader_simulator(self.exchange):
+            await self._create_order_if_enabled(self.symbol_evaluator.get_trader(self.exchange),
+                                                notification_candle_time)
+
+        if self.symbol_evaluator.has_trader(self.exchange):
+            await self._create_order_if_enabled(self.symbol_evaluator.get_trader_simulator(self.exchange),
+                                                notification_candle_time)
+
+    async def _create_order_if_enabled(self, trader, notification_candle_time):
+
+        if trader.is_enabled():
+            # cancel previous by orders if any
+            cancelled_orders = await self._cancel_buy_orders_for_trader(trader)
+            if self.last_buy_candle == notification_candle_time and cancelled_orders or \
+                    self.last_buy_candle != notification_candle_time:
+                # if subsequent notification from the same candle: only create order if able to cancel the previous buy
+                # to avoid multiple order on the same candle
+                await self._create_order(trader)
+                self.last_buy_candle = notification_candle_time
+            else:
+                self.logger.debug(f"Ignored buy signal for {self.symbol}: buy order already filled.")
+
+    async def _create_order(self, trader, buy=True, total_sell=None, sell_target=None, buy_price=None):
```

Comment:
Should this implementation be added in AbstractTradingModeDecider ?

Reply:
I don't think it's a good idea, ```total_sell=None, sell_target=None, buy_price=None``` params are specific to this trading mode, same for ```_create_buy_order``` and ```_create_sell_orders``` methods

Reply:
Ok !